FilecoinFoundationWeb/filecoin-foundation
# Governance Page - Upcoming Events section

## Changes
- new `GovernanceCalendarCard` component
- new `convertToISOTime` utils file

## How to test
- test if clicking on "Add Event to Calendar" allows you to add the event to your calendar. Also check if the time matches the time stated in the `GovernanceCalendarCards`

## Note
1.  you can find the` NEXT_PUBLIC_CALENDAR_ID` and `NEXT_PUBLIC_CALENDAR_API_KEY` for `.env.local` in 1password, under UXIT team
2. Why `GovernanceCalendarCards` takes `currentDate` as a prop?
- in order to get all future dates I've used query param `timeMin` which takes ISO date as a value. Since I wanted to use today's date I've used `new Date()` method , however that caused some issues with the API call since the Date object hasn't been fully initialised by the time the URL is constructed, leading to this error:
`Uncaught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received`
To prevent that I passed the `currentDate` as a prop, and that seemed to work. If you think it could be done in a better way, let me know!


## Visuals

![Screenshot 2024-05-14 at 15 50 36](https://github.com/FilecoinFoundationWeb/filecoin-foundation/assets/50910606/742b0323-cf33-4791-90ee-98643c65d052)
![Screenshot 2024-05-14 at 15 50 29](https://github.com/FilecoinFoundationWeb/filecoin-foundation/assets/50910606/697e3973-9d99-4866-becb-1046223b6f08)
![Screenshot 2024-05-14 at 15 50 23](https://github.com/FilecoinFoundationWeb/filecoin-foundation/assets/50910606/9a5c7216-e39d-45e2-836f-2bedc1f85ab9)
